packages/block-library/src/separator/deprecated.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { omit } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { getColorClassName, useBlockProps } from '@wordpress/block-editor';
+
+const v1 = {
+	attributes: {
+		color: {
+			type: 'string',
+		},
+		customColor: {
+			type: 'string',
+		},
+	},
+	save( { attributes } ) {
+		const { color, customColor } = attributes;
+
+		// the hr support changing color using border-color, since border-color
+		// is not yet supported in the color palette, we use background-color
+		const backgroundClass = getColorClassName( 'background-color', color );
+		// the dots styles uses text for the dots, to change those dots color is
+		// using color, not backgroundColor
+		const colorClass = getColorClassName( 'color', color );
+
+		const className = classnames( {
+			'has-text-color has-background': color || customColor,
+			[ backgroundClass ]: backgroundClass,
+			[ colorClass ]: colorClass,
+		} );
+
+		const style = {
+			backgroundColor: backgroundClass ? undefined : customColor,
+			color: colorClass ? undefined : customColor,
+		};
+
+		return <hr { ...useBlockProps.save( { className, style } ) } />;
+	},
+	isEligible( { opacity } ) {
+		return ! opacity;
```

Comment:
Apologies, my memory here might be a little foggy, but do we need to check for the presence of the `opacity` attribute if we know that the markup will be different (e.g. in the new version, there'll be a `has-alpha-channel-opacity` class?)

Reply:
Because the likes of 
```
<!-- wp:separator -->
<hr class="wp-block-separator"/>
<!-- /wp:separator -->
```
does not invalidate in the current save method we have to use the isEligible to force it through the migration